src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Subscriptions/DependenciesSnapshotProvider.cs

```diff
@@ -289,6 +292,11 @@ private void DisposeCore()
                     projectItemSpecs),
                 token);
 
+            if (updatedSnapshot != null)
```

Comment:
This could result in us sending an incomplete snapshot right? ie You do a big branch switch, close VS before the tree has finished we might send unresolved states that are misleading (ie its just that we never received design-time build data).

Perhaps we should only update the snapshot state when we're up-to-date with the latest design-time build data?

Reply:
No, the behaviour is the same as before. `updatedSnapshot` is only non-`null` if the `TryUpdate` method actually made a change in response to the update.

Reply:
I think its more than when we get a JointRule; we can get inbetween states that might fixed themselves once we've caught up. For example, go change the version number a commonly used package in Package.targets to see this - you'll a bunch of errors that will eventually go away, you don't ever want to report a telemetry event during that because we'll get misleading data (if had let it finish, they would have been resolved).

I think we'll want to make sure that we're up-to-date with the latest version of JointRule or perhaps operation progress. @lifengl Thoughts?